src/app/services/topic.service.ts

```diff
@@ -1,9 +1,232 @@
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+      ],
+    },
+    {
+      name: 'JAVA',
+      subtopic: [
+        {
+          title: 'What is JAVA?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+        {
+          title: 'What is Lorem Ipsum?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+        {
+          title: 'What is Lorem Ipsum?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+        {
+          title: 'What is Lorem Ipsum?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+      ],
+    },
+  ];
 
-  constructor() { }
+  constructor() {}
+
+  getListOfTopics(): Topic[] {
+    return this.TOPICSLIST;
+  }
+
+  getListOfSubtopicsByTopicName(topic: string): SubTopic[] | undefined {
+    return this.TOPICSLIST.filter((el) => el.name === topic)[0]?.subtopic;
+  }
+
+  getSubtopicBySearchedValue(
+    searchedValue: string,
+    topicName: string
+  ): SubTopic[] | undefined {
+    const listOfSubtopics = this.getListOfSubtopicsByTopicName(topicName);
+    return listOfSubtopics?.filter(
+      (el) =>
+        el.title.toLowerCase().includes(searchedValue) ||
```

Comment:
searchedValue should also be toLowerCase(). Do it only once and reuse that value from a variable

Reply:
I did it here: 
![image](https://github.com/radumaftei/frontend-learning-app/assets/100227050/be40d763-90fd-43f1-9d52-8a3eb8fbc22b)

Reply:
Call ? Yes?